probe-rs/src/probe/espusbjtag/mod.rs

```diff
@@ -484,9 +486,16 @@ impl DebugProbe for EspUsbJtag {
         let taps = self.scan()?;
         tracing::info!("Found {} TAPs on reset scan", taps.len());
 
-        let selected = 0;
+        let mut selected = 0;
         if taps.len() > 1 {
-            tracing::warn!("More than one TAP detected, defaulting to tap0");
+            // TODO: https://github.com/probe-rs/probe-rs/issues/2058
+            // There is no way to select the tap currently
+            // The HP cores of the ESP32-P4 are 2nd in the chain,
+            // therefore we default to selecting tap1
+            if taps.iter().any(|x| x.idcode == Some(IdCode(0x0012c25))) {
```

Comment:
Is this just for testing, or will you include this piece of loveliness in every probe driver?

Reply:
I don't have a solution, this was my ugly hack until we figure something out in #2058, open to suggestions :).